### ISSUE TITLE
Add support for module.extensions

### ISSUE THREAD

**Issue by simast** [igorklopov/enclose#83]
While not documented in the public docs - Node's module system has a very powerful method of adding custom extension support for `require()` calls.

For example:

``` javascript
const fs = require("fs");
const JSON5 = require("json5");

require.extensions[".json5"] = (module, file) => {
    module.exports = JSON5.parse(fs.readFileSync(file, "utf8"));
};
```

Now I can use `require("./test")` to load `test.json5` file as a module. This will confuse enclose however, as it will look only for the default extensions when compiling (`.js`, `.json` and `.node`).

Could the `scripts` config file property be extended to support including custom extension modules like that?

``` javascript
module.exports = {
    scripts: [
        "**/*.json5"
    ]
};
```

**Comment by igorklopov** [comment]
@simast Honestly, your idea is not totally clear to me, but i advice you to try `module.extensions` with just released enclose 2.0.0. There was a significant refactor. Probably you may get some results. Anyway please post all your feedback in detail. What you did, what you expected (say, original node works this way) and what you get. Thanks!

**Comment by igorklopov** [comment]
Dear users of EncloseJS! I highly encourage you to switch to https://github.com/zeit/pkg
It is rewritten successor of EncloseJS. It is open source, and all improvements will go there.
Please check if pkg has your issue unresolved, and if it is true, reopen your issue there:
https://github.com/zeit/pkg/issues
